execution-core/src/transfer/phoenix.rs

```diff
@@ -246,22 +246,22 @@ impl Transaction {
         let schnorr_sk_b = SchnorrSecretKey::from(sender_sk.b());
         let sig_b = schnorr_sk_b.sign(rng, payload_hash);
 
-        let circuit = TxCircuitVec {
-            input_notes_info,
-            output_notes_info,
-            payload_hash,
-            root,
-            deposit,
-            max_fee,
-            sender_pk,
-            signatures: (sig_a, sig_b),
-        };
-
         Self {
             payload,
-            proof: P::prove(circuit).expect(
-                "The proof generation shouldn't fail with a valid circuit",
-            ),
+            proof: P::prove(
+                &TxCircuitVec {
+                    input_notes_info,
+                    output_notes_info,
+                    payload_hash,
+                    root,
+                    deposit,
+                    max_fee,
+                    sender_pk,
+                    signatures: (sig_a, sig_b),
+                }
+                .to_var_bytes(),
+            )
+            .expect("The proof generation shouldn't fail with a valid circuit"),
```

Comment:
As discussed, it's better to have a recoverable error here instead of a panic.

Reply:
Should be solved now